### ISSUE TITLE
Polish first launch dialog

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#4531]
See [mockup](https://www.figma.com/file/SkyAE9BZVmPQYnrPb30peg/First-launch-experience?node-id=104%3A00) discussed in [forum thread](https://forum.getodk.org/t/collect-prompt-for-server-configuration-on-first-launch/33051) and by TAB.

- [x] Update first launch screen following rough guidance to start with:
- Use outline edit icon
- 60x60 [ODK logo](https://getodk.org/assets/zip/odk-brand-assets.zip)
- 20sp button text, medium weight (got feedback that initial mockups were too small/light)
- 35sp tagline text
- 18sp version text
- 14sp "try Collect" text
- My general sense of the middle part is that it's all in a container that hugs contents. The width is determined by the width of the widest button. The two buttons are the same width. The whole container is centered horizontally and vertically.
- I don't know how the button icon and text should be laid out (centered, left aligned). I'm hoping there's a sensible default that can just be used.
- I also don't really know about margins and vertical spacing between elements. I'm hoping there are also some defaults that can be relied on there.

These are all starting points. We should check in about anything that's hard and see if there are easier alternatives. We'll likely want to do another pass of adjusting font sizes once we can look at it on different screens.

- [x] Change first launch screen to `Activity`. The `SplashScreenActivity` will be going away soon. There's some glitchy navigation and backstack manipulation going on right now. (https://github.com/getodk/collect/issues/4531#issuecomment-858689395)

- [x] Leave QR code config dialog when manual config dialog is launched from it instead of manipulating the back stack. (https://github.com/getodk/collect/issues/4531#issuecomment-861619917)

**Comment by lognaturel** [comment]
Consider making this an `Activity` instead of a `DialogFragment` for performance reasons. Try on worst phone to experience performance degradation caused by animations. Also, scanning from QR code should directly go to MainMenuActivity instead of showing first launch dialog briefly. Settings import should happen off UI thread.

From https://github.com/getodk/collect/issues/4439#issuecomment-857813161: make sure we can scroll horizontally

**Comment by lognaturel** [comment]
@grzesiek2010 @seadowg I'm going to start taking a look at this over the next few days while y'all are wrapping up other beta things. Hopefully I'll be able to finish it but if one of you completes everything in your pipeline and I'm not done, let's talk about transferring it to you.

**Comment by lognaturel** [comment]
I've successfully made the first launch screen into an `Activity` and that definitely makes more sense to me. However, I can't figure out how to prevent that screen from showing briefly when a project is successfully created and the project creation dialog is dismissed. Any ideas, @grzesiek2010, @seadowg? See https://github.com/getodk/collect/pull/4601/files/33b9cdbc0dc7fefb4fe8b00b69046b5492f7fb8c#diff-78a8fdb05c2a21720fa00d2097e76a82f212c91fa10693d078bd6c9338b0c601R141

**Comment by grzesiek2010** [comment]
> However, I can't figure out how to prevent that screen from showing briefly when a project is successfully created and the project creation dialog is dismissed. Any ideas

Maybe just delay dismissing that dialog. I did something like that in the same class https://github.com/getodk/collect/pull/4601/files/33b9cdbc0dc7fefb4fe8b00b69046b5492f7fb8c#diff-78a8fdb05c2a21720fa00d2097e76a82f212c91fa10693d078bd6c9338b0c601L64
The problem was the same.

**Comment by seadowg** [comment]
@lognaturel  delaying could work, but it looks like you can just remove the `dismiss` call now that you are opening the main menu from the dialog. I'm also wondering if we should be using [`CLEAR_TOP`](https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_CLEAR_TOP) instead of `finishAffinity`. I'd never seen `finishAffinity` before and looking into it does seem people have complained about visual quirks before (SO [here](https://stackoverflow.com/questions/33497151/activity-finishaffinity-vs-intent-flag-activity-new-task-intent-flag-activit))

**Comment by lognaturel** [comment]
Thanks!

Removing the dismiss call does lead to the experience we want but wouldn’t it lead to a leak in the QR code case? Or does onDestroy get called on the fragment get called if it’s called on the activity?

I didn’t try CLEAR_TOP because I saw there might be different behavior by Android version but I should check what API levels!

**Comment by seadowg** [comment]
>  Or does onDestroy get called on the fragment get called if it’s called on the activity?

It really should, as destroying the activity should destroy all its Fragments. Maybe good to verify with debugger just in case?

> I didn’t try CLEAR_TOP because I saw there might be different behavior by Android version but I should check what API levels!

Of course 😩

**Comment by lognaturel** [comment]
Thanks, both! I wasn't able to find Android docs saying that `Fragment` lifecycle methods are called when `Activity` methods are but it certainly would make the most sense. [This course](https://google-developer-training.github.io/android-developer-advanced-course-concepts/unit-1-expand-the-user-experience/lesson-1-fragments/1-2-c-fragment-lifecycle-and-communications/1-2-c-fragment-lifecycle-and-communications.html) which I think was published by Google explicitly spells that out and I verified it with the debugger. So I've removed `dismiss`.

I also made another behavior change: I left the QR code configuration dialog on screen when the manual configuration dialog is invoked from it. That means if a user taps to configure by QR code, then wants to see what manual config is about, then cancels manual config, they go back to QR code config. This makes the most sense to me and in general I don't think we should be changing the backstack unless certain screens like the first launch screen are never to be visited again. I'll write this out in the PR but wanted to capture the high level reasoning here as well.
